app/flatpak-builtins-search.c

```diff
@@ -0,0 +1,221 @@
+          g_clear_error (&error);
+          continue;
+        }
+      else if (remotes == NULL)
+        continue;
+
+      for (guint j = 0; remotes[j]; ++j)
+        {
+          g_autofree char *appstream_file = g_build_filename (install_path, "appstream", remotes[j],
+                                                              flatpak_get_arch(), "active", "appstream.xml.gz",
+                                                              NULL);
+          g_ptr_array_add (ret, g_file_new_for_path (appstream_file));
+        }
+    }
+  return ret;
+}
+
+static AsStore *
+store_new (GCancellable *cancellable)
+{
+  AsStore *store = as_store_new ();
+  g_autoptr(GPtrArray) appstream_dirs = get_appstream_dirs (cancellable);
+
+  for (guint i = 0; i < appstream_dirs->len; ++i)
+    {
+      GFile *appstream_file = g_ptr_array_index (appstream_dirs, i);
+      g_autoptr(GError) error = NULL;
+      as_store_from_file (store, appstream_file, NULL, cancellable, &error);
+      if (error)
+        {
+          // We want to ignore this error as it is harmless and valid
+          // NOTE: appstream-glib doesn't have granular file-not-found error
+          if (!g_str_has_suffix (error->message, "No such file or directory"))
+            g_warning ("%s", error->message);
+        }
+    }
+
+  return store;
+}
+
+typedef struct MatchResult {
+  AsApp *app;
+  guint score;
+} MatchResult;
+
+static void
+match_result_free (MatchResult *result)
+{
+  g_object_unref (result->app);
+  g_free (result);
+}
+
+static MatchResult *
+match_result_new (AsApp *app, guint score)
+{
+  MatchResult *result = g_new (MatchResult, 1);
+  result->app = g_object_ref (app);
+  result->score = score;
+  return result;
+}
+
+static int
+compare_by_score (MatchResult *a, MatchResult *b, gpointer user_data)
+{
+  // Reverse order, higher score comes first
+  return (int)b->score - (int)a->score;
+}
+
+static const char *
+get_comment_localized (AsApp *app)
+{
+  const char * const * languages = g_get_language_names ();
+  for (gsize i = 0; languages[i]; ++i)
+    {
+      const char *comment = as_app_get_comment (app, languages[i]);
+      if (comment != NULL)
+        return comment;
+    }
+  return NULL;
+}
+
+static void
+print_app (MatchResult *res, FlatpakTablePrinter *printer)
+{
+  /* TODO:
+   * - AsApp doesn't expose the flatpak repository name
+   */
+  AsRelease *release = as_app_get_release_default (res->app);
+  const char *version = release ? as_release_get_version (release) : NULL;
+
+  flatpak_table_printer_add_column (printer, as_app_get_id_filename (res->app));
+  flatpak_table_printer_add_column (printer, version);
+  flatpak_table_printer_add_column (printer, get_comment_localized (res->app));
+  flatpak_table_printer_finish_row (printer);
+}
+
+gboolean
+flatpak_builtin_search (int argc, char **argv, GCancellable *cancellable, GError **error)
+{
+  g_autoptr(GOptionContext) context = g_option_context_new (_("TEXT - Search for text"));
```

Comment:
You could mention that it's searching the remotes (as opposed to searching locally installed apps/runtimes)